contracts/debt/TokemakIncurDebt.sol

```diff
@@ -0,0 +1,305 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity ^0.8.4;
+
+import "../interfaces/IOHM.sol";
+import "../interfaces/IsOHM.sol";
+import "../interfaces/IgOHM.sol";
+import "../interfaces/IERC20.sol";
+import "../interfaces/IStaking.sol";
+import "./interfaces/ITreasury.sol";
+import "../libraries/SafeERC20.sol";
+import "../interfaces/IIncurDebt.sol";
+import "../types/OlympusAccessControlledV2.sol";
+
+contract TokemakIncurDebt is OlympusAccessControlledV2 {
+    using SafeERC20 for IERC20;
+    uint256 public globalDebtLimit;
+
+    address public immutable OHM;
+    address public immutable gOHM;
+    address public immutable sOHM;
+    address public immutable staking;
+    address public immutable treasury;
+
+    struct Borrower {
+        uint256 debt;
+        uint256 limit;
+        uint256 collateralInSOHM;
+        uint256 collateralInGOHM;
+        bool isBorrower;
+    }
+
+    mapping(address => bool) isStrategy;
+    mapping(address => Borrower) borrower;
+
+    constructor(
+        address _OHM,
+        address _gOHM,
+        address _sOHM,
+        address _staking,
+        address _treasury,
+        address _olympusAuthority
+    ) OlympusAccessControlledV2(IOlympusAuthority(_olympusAuthority)) {
+        OHM = _OHM;
+        gOHM = _gOHM;
+        sOHM = _sOHM;
+        staking = _staking;
+        treasury = _treasury;
+    }
+
+    /**
+     * @notice sets the maximum debt limit for the system
+     * - onlyOwner (or governance)
+     * - must be greater than or equal to existing debt
+     * @param _limit in OHM
+     */
+    function setGlobalDebtLimit(uint256 _limit) external onlyGovernor {
+        require(_limit >= globalDebtLimit);
```

Comment:
We should be able to drop the limit down. We should keep track of the total outstanding debt and make sure we can't go below that number.

Reply:
Done